From 842bdfbb622db0673d4952812a490e10f61fa3bd Mon Sep 17 00:00:00 2001
Message-Id: <842bdfbb622db0673d4952812a490e10f61fa3bd.1527784094.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Thu, 31 May 2018 17:01:16 +0200
Subject: [PATCH 1/2] makepkg: Clear ERR trap before trying to restore it

$restoretrap is empty if the trap was not set. This caused the trap
handler to remain and override later exit codes.
---
 scripts/makepkg.sh.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/makepkg.sh.in b/scripts/makepkg.sh.in
index e9080a70..d67fd853 100644
--- a/scripts/makepkg.sh.in
+++ b/scripts/makepkg.sh.in
@@ -432,6 +432,7 @@ run_function_safe() {
 
 	run_function "$1"
 
+	trap - ERR
 	eval "$restoretrap"
 	eval "$restoreset"
 	eval "$restoreshopt"
-- 
2.17.0
